ipc/provider/src/checkpoint.rs

```diff
@@ -151,21 +169,41 @@ impl<T: BottomUpCheckpointRelayer + Send + Sync + 'static> BottomUpCheckpointMan
     async fn submit_next_epoch(&self, submitter: &Address) -> Result<()> {
         let next_submission_height = self.next_submission_height().await?;
```

Comment:
Just a sanity-check: remember that relayers are allowed to submit a checkpoint for height `h` even if it has been submitted already as long as no submission have been made yet for `h+1`

Reply:
yep, there is a submit last method